### ISSUE TITLE
Compression?

### ISSUE THREAD

**Issue by VeniceNerd** [rcodehub/unraid-plex-script#2]
Hi! This is a very cool script and I think I’ll give this a try. Just wondering if the backup will be compressed at all. Would be really cool to have an option for that!

**Comment by rcodehub** [comment]
Currently it does not use compression.  I opted for 'speed' over compression as the metadata could be extremely large and time consuming to archive.

It would be easy enough to add compression to the script by modifying the tar command on each line that creates an archive.

You can experiment using different levels of compression by using the information here: https://www.gnu.org/software/tar/manual/tar.html#Compression

A brief test of my essential data without compression was 32 seconds, 4.86GB and with lzma compression using tar -caf and changing the extension from .tar to .tar.lzma was 266 seconds with the same file size... so no point in doing that. ;)

**Comment by VeniceNerd** [comment]
Copy that! Do you think it would be worth it with the full backup set, though?

**Comment by rcodehub** [comment]
It's hard to say.  I'm sure there would be some amount of compression 
but is it worth the additional time that it would take for the 
compression?  I suppose that is a personal preference.

Playing with the tar compression levels is one option, or using a 
completely different compressor might have better results.

p7zip looks like it might be fun to experiment with but it would require 
the package installation from the NerdPack plugin making this not a 
standalone script.

When I have some time to play around with this I'll try some benchmarks 
and that might lead to an update to this script.

**Comment by VeniceNerd** [comment]
Oh cool! If you do keep me updated, please! Already love this script as is!
________________________________

**Comment by rcodehub** [comment]
It actually didn't take long to test this out.  I tried a couple of the tar options with no real difference in file size, only increases in time, so I wouldn't use tar for trying to utilize compression.

I tried p7zip and tested the difference on a full backup of my data with maximum compression.

Original = 419 seconds, approximately 7 minutes, 28.2GB
7z = 2560 seconds, approximately 43 minutes. 21.4GB

I personally don't feel that 7gb is worth an additional 36 minutes of backup time, but if you do I can walk you through modifying the script.
